src/eradiate/experiments/_helpers.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import annotations
+
+from ..scenes.bsdfs import BSDF, bsdf_factory
+from ..scenes.measure import MultiRadiancemeterMeasure
+from ..scenes.measure._distant import DistantMeasure
+from ..scenes.shapes import RectangleShape
+from ..scenes.surface import BasicSurface, surface_factory
+
+# ------------------------------------------------------------------------------
+#                             Experiment helpers
+# ------------------------------------------------------------------------------
+
+def measure_inside_atmosphere(atmosphere, measure, ctx):
+    """
+    Evaluate whether a sensor is placed within an atmosphere.
+
+    Raises a ValueError if called with a :class:`.MultiRadiancemeterMeasure`
+    with origins both inside and outside the atmosphere.
+    """
+    if atmosphere is None:
+        return False
+
+    shape = atmosphere.eval_shape(ctx)
+
+    if isinstance(measure, MultiRadiancemeterMeasure):
+        inside = shape.contains(measure.origins)
+
+        if all(inside):
+            return True
+        elif not any(inside):
+            return False
+        else:
+            raise ValueError(
+                "Inconsistent placement of MultiRadiancemeterMeasure origins. "
+                "Origins must lie either all inside or all outside of the "
+                "atmosphere."
+            )
+
+    elif isinstance(measure, DistantMeasure):
+        # Note: This will break if the user makes something weird such as using
+        # a large offset value which would put some origins outside and others
+        # inside the atmosphere shape
+        return not measure.is_distant()
+
+    else:
+        # Note: This will likely break if a new measure type is added
+        return shape.contains(measure.origin)
+
+
+def _surface_converter(value):
```

Comment:
Since this converter is no longer close to the field it is used for, I think a brief docstring would be good.